src/eval.rs

```diff
@@ -221,7 +221,10 @@ pub fn eval_main<'tcx>(tcx: TyCtxt<'tcx>, main_id: DefId, config: MiriConfig) {
                 };
 
                 let msg = format!("Miri evaluation error: {}", msg);
-                let mut err = struct_error(ecx.tcx.tcx.at(span), msg.as_str());
+                let mut err = {
+                    let new_tcx = ecx.tcx.tcx.at(span);
+                    new_tcx.sess.struct_span_err(new_tcx.span, msg.as_str())
```

Comment:
```suggestion
                    ecx.tcx.sess.struct_span_err(span, msg.as_str())
```

Reply:
yea, the old `tcx.at` dance was just because of the `struct_error` API, and is not really something that is relevant for `struct_span_err`